pkg/kubecost/service/allocation/allocationsummary.proto

```diff
@@ -0,0 +1,147 @@
+syntax = "proto3";
+
+package kubecost.service.allocation.allocationsummary;
+
+option go_package = "github.com/opencost/opencost/pkg/kubecost/service/allocation/allocationsummary";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+// Allocation Summary Interface exported by the gRPC server.
+service AllocationSummary {
+    // An RPC for allocation.
+    //
+    // Obtains the allocation summary for a given window span.
+    
+    // Obtains the AllocationSummarySets satisfy the given AllocationSummaryRequest. Results are
+    // streamed rather than returned at once.
+    rpc GetAllocationSummary (AllocationSummaryRequest) returns (stream AllocationSummaryResponse) {}
+}
+
+// AllocationSummaryRequest is the request send from client with a specify window
+// either absolute window with datetime or relative window with duration
+message AllocationSummaryRequest {
+    Window window =1;
+    repeated string aggregate =2;
+    bool accumulate=4;
+    FilterSelection filter = 5;
+    SharedSelection shared = 6;
+    IdleSelection idle = 7;
+    bool external =8;
+}
```

Comment:
add a 

// The maximum number of items to return in the table. If unspecified/set to 0, we will return all items.
int32 max_table_items = 0;

Reply:
Minor clarification , the 0 after the name is not possible as field number https://developers.google.com/protocol-buffers/docs/proto3#assigning_field_numbers. I'm assuming u only meant default to 0 here.